go/ethadapter/blockprovider.go

```diff
@@ -207,12 +216,31 @@ func (e *EthBlockProvider) awaitNewBlock(ctx context.Context) (*types.Header, er
 			if errors.Is(err, ErrSubscriptionNotSupported) {
 				return nil, err
 			}
-			e.logger.Error("L1 block monitoring error", log.ErrKey, err)
+			e.logger.Error("L1 block monitoring error - Restarting L1 block Monitoring...", log.ErrKey, err)
+
+			if !e.ethClient.Alive() {
```

Comment:
Maybe worth moving this block to be an e.ensureClientConnected() method or something since it's duplicated below?